### PR TITLE
Diff state at stage and commit pages

### PR DESCRIPTION
**Pull request by sbcgua**
Stage page
2019-10-25_22h55_40
Commit page
2019-10-25_22h55_52

P.S. review welcomed, imho a bit crutchy, but not sure how to do better without major refactoring
P.P.S `apply_order_by` and `render_order_by` unchanged - reordrings

### CODE REVIEW

**Comment by christianguenter2** [comment]
I did some tests and didn't spot any issues. Nice feature, btw.

File: src/ui/zcl_abapgit_gui_page_stage.clas.abap [hunk 1]
```diff
@@ -252,6 +254,12 @@ CLASS ZCL_ABAPGIT_GUI_PAGE_STAGE IMPLEMENTATION.
           ev_path     = ls_file-path
           ev_filename = ls_file-filename ).
 
+      READ TABLE ms_files-status ASSIGNING <ls_status>
+        WITH KEY
+          path     = ls_file-path
+          filename = ls_file-filename.
```
Comment (christianguenter2) [inline src/ui/zcl_abapgit_gui_page_stage.clas.abap]: Suggest using `WITH TABLE KEY` to optimize key access as it's a hashed table with keys path and filename. Same for others usages below.

Comment (sbcgua) [reply src/ui/zcl_abapgit_gui_page_stage.clas.abap]: adjusted, though seems to be unnecessary  (see comments below)

**Comment by sbcgua** [comment]
Regarding table key - sap seems to auto identify this. Made some test:

2019-10-26_15h02_41

Source is here: https://github.com/sbcgua/benchmarks

P.S. surprise ! linear+binary faster than native sorted tab, strange, does not make sense - it should be the same approach ...

File: src/ui/zcl_abapgit_gui_page_stage.clas.abap (hunk 1 above)
Comment (christianguenter2) [reply src/ui/zcl_abapgit_gui_page_stage.clas.abap]: Interesting! I thought that omitting `TABLE` always leads to a linear search for hash tables. Obviously I was wrong. Thanks.
https://help.sap.com/doc/abapdocu_752_index_htm/7.52/en-US/abapread_table_free.htm
image

**Comment by christianguenter2** [comment]
@sbcgua Do you think that feature should also work for patch? I'm not sure, but it would be nice.
Currently it looks like this. But we could do this also in a follow up PR.
image

**Comment by sbcgua** [comment]
hmmm ... I don't know patch codebase well ... but I'll have a look ... should be as easy as inserting one read table somewhere hopefully

**Comment by sbcgua** [comment]
@christianguenter2 
> ... should be as easy as inserting one read table somewhere hopefully ...

as usual after a phrase like this ... it is not that easy. :) 

lets imagine an example of MM file (modified from both sides)
What if I select only those lines modified locally  ? Should the status in commit look like MM or M_ ? And will this work correctly at all (I don't use patch a lot - I don't know)
Also I'd do some refactorings in diff page first. E.g. lo_state is created at stage page but it's empty and not returned there, so I'd move creation into diff page and some other minor stuff
So to summarize I propose to make it in another PR later. I have some more small improvements to stage page to implement soon.

**Comment by christianguenter2** [comment]
Yeah agree. Let's postpone that. I already guessed that it wouldn't be too easy.
